### PR TITLE
271 Add cloud sql auth proxy

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context

We need a cloudsql proxy image.

Also re-done the folder naming to be more consistent with the image names, since this has caused some confusion.

# What has changed

- Add cloudsql proxy dockerfile
- Update Makefile and README
- Update paths to be more consistent with naming

# How to test?

Try building the image and using it.

# Links

https://trello.com/c/nI77Ukwq/271-add-cloud-sql-auth-proxy-to-bastion-vm-5

### CODE REVIEW

File: cloudsql-proxy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM eu.gcr.io/cloudsql-docker/gce-proxy:1.31.2-alpine
+EXPOSE 5432
+USER root
+RUN apk update && apk add --no-cache postgresql-client
```
Comment (davidmort) [inline cloudsql-proxy/Dockerfile:4]: whats the reason for installing this?

**Review by davidmort — CHANGES_REQUESTED** [review]

**Review by davidmort — APPROVED** [review]

**Comment by apolloakora** [comment]
We should document the github workflow where it logs into dockerhub but that is not in the scope of this - just noticed that.

File: cloudsql-proxy/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FROM eu.gcr.io/cloudsql-docker/gce-proxy:1.31.2-alpine
```
Comment (apolloakora) [inline cloudsql-proxy/Dockerfile:1]: When I ran docker scan on the cloudsql-proxy image it produced a critical severity in zlib (Out of bounds write) which it said was fixed in zlib 1.2.12-r2

**Review by apolloakora — CHANGES_REQUESTED** [review]

File: cloudsql-proxy/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+EXPOSE 5432
+USER root
+RUN addgroup --gid 1000 cloudsqlproxy && adduser --system --uid 1000 cloudsqlproxy cloudsqlproxy
+USER cloudsqlproxy
+# CLOUDSQL_INSTANCE_ADDRESS format: <PROJECT>:<CLOUDSQL_INSTANCE_ID>=tcp:<PORT>
+CMD ./cloud_sql_proxy -enable_iam_login -instances=$CLOUDSQL_INSTANCE_ADDRESS -ip_address_types=PRIVATE
```
Comment (apolloakora) [inline cloudsql-proxy/Dockerfile:7]:
```suggestion
CMD ./cloud_sql_proxy -enable_iam_login -instances=$CLOUDSQL_INSTANCE_ADDRESS -ip_address_types=PRIVATE

```

**Review by apolloakora — CHANGES_REQUESTED** [review]

File: cloudsql-proxy/Dockerfile (hunk 2 above)
Comment (ryangrundy7) [reply cloudsql-proxy/Dockerfile:1]: I don't think there's much we can do about this. It's the latest version of the dockerfile so there's no current fix for it

**Review by apolloakora — APPROVED** [review]

File: cloudsql-proxy/Dockerfile (hunk 2 above)
Comment (davidmort) [reply cloudsql-proxy/Dockerfile:1]: reading up on that zlib cve it doesn't look like we'd be affected by it anyway 👍

**Pull request merged**